CodeU_Assignement_1/src/codeu_assignement_1/SinglyLinkedList.java

```diff
@@ -0,0 +1,236 @@
+    /*  Check if the list is empty  */
+    public boolean isEmpty() { return (m_first == null); }
+
+    /*  Get the current size of the list  */
+    public int getSize() { return size; }
+ 
+    /*  Add e element in the front/begining of the list  */
+    public void addFirst(E e) { addAt(0,e); }
+    
+    /*  Add e element in the tail/end of the list  */
+    public void addLast(E e) { addAt(size,e); }
+    
+    /*  Add e element at index  */
+    public void addAt(int index, E e)
+    {        
+        // check - index is begger than size or neg.
+        if (index < 0 || index > size)
+            throw new IndexOutOfBoundsException();
+     
+        Node n = new Node(e, null);             
+                
+        // Empty list
+        if(m_first == null) 
+        {
+            m_first = n;
+            m_last = m_first;
+        }
+        // Add to first
+        else if (index == 0)
+        {
+            n.setNext(m_first);
+            m_first = n;
+        }
+        // Add to last
+        else if (index == size)
+        {
+            m_last.setNext(n);
+            m_last = n;
+        }
+        // Add to in the middle
+        else 
+        {
+            Node curr = m_first;
+            index -= 1; // i = i-1; to get the item befoer where we add
+            for (int i = 0; i < size; i++) 
+            {
+                if (i == index) 
+                {
+                    // Before: curr -> next_curr
+                    Node next_curr = curr.getNext() ;
+                    curr.setNext(n);
+                    n.setNext(next_curr);
+                    break;
+                    // After: curr -> n -> next_curr
+                }
+
+                // Go to next
+                curr = curr.getNext();
+            }
+        }
+        
+        size++ ;
+    }
+    
+    /*
+        Add all the items in the collection into the list
+        NullPointerException - if the specified collection is null
+    */
+    public boolean addAll(Collection<? extends E> c)
+    {
+        if (c == null)
+            throw new NullPointerException();
+        
+        Iterator <? extends E> itr = c.iterator();
+        if (!itr.hasNext()) 
+            return false;
+        
+        // If there are items - add them one after another
+        while (itr.hasNext()) 
+        {
+            this.addLast(itr.next());
+        }
+        
+        return (size > 0); // Shall add something
+    }
+    
+    /*  Add e element in the front/begining of the list  */
+    public E removeFirst() { return removeAt(0); }
+    
+    /*  Add e element in the tail/end of the list  */
+    public E removeLast() { return removeAt(size-1); }
+    
+    /*  Add e element at index  */
+    public E removeAt(int index)
+    {        
+        // check - index is begger than size or neg.
+        if (index < 0 || index >= size)
+            throw new IndexOutOfBoundsException();
+        
+        if (index == 0) // Remove first 

```

Comment:
In this case you can only remove index 0 (else 156 line exception).
m_first will be set to null, the size will be set to 0 and ret will contain the only element that was in the list. I checked it with debugger with this code:

            SinglyLinkedList<Integer> my_list;
            my_list = new SinglyLinkedList<>();
            my_list.addFirst(1);
            my_list.removeFirst();

Is this the case you asked about?